### PR TITLE
feat: overhaul open trade fees

### PR DESCRIPTION
**Pull request by MeLlamoPablo**
This PR:

* Considers the "deposit and trade" fee and the "trade" fee on the open trade flow. These fees are not yet enabled on the SC side.
* Refactors the "open trade" amount calculations in `TradeStore` by extracting them to a separate `open-trade-utils` file.

    This refactor aims to make clearer the flow of operations applied to the user input:

    * If paying with ETH:
        * Quote ETH -> wstETH via Balancer
        * Apply deposit slippage to the quote
        * Apply prePO fees to the quote
    * If paying with collateral:
        * Convert the user ETH input into wstETH
        * Apply prePO fees to the value

    I still think that `TradeStore` values can be made less confusing, but IMO this is a good start.

### CODE REVIEW

File: apps/frontend/core-dapp/src/features/trade/open-trade-utils.ts [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+  const depositAndTradeFeeBN = amountAfterDepositSlippageInWstEth
+    .mul(depositAndTradeFee)
+    .div(depositTradeHelperPercentDenominator)
+
+  const depositFeeBN = amountAfterDepositSlippageInWstEth
+    .mul(depositFee)
+    .div(collateralPercentDenominator)
+
+  return depositAndTradeFeeBN.add(depositFeeBN)
```
Comment (MeLlamoPablo) [inline apps/frontend/core-dapp/src/features/trade/open-trade-utils.ts:113]: @XavierEkkel this is the relevant fee logic

**Review by XavierEkkel — COMMENTED** [review]
please split up PR into single-purpose PRs

**Review by chrisling-dev — COMMENTED** [review]
The approach in this PR is great, I really like it. Leaving a comment without approval for now as I need more time to review and test the changes.

**Comment by MeLlamoPablo** [comment]
Superseded by #1144 and #1145

**Pull request closed without merging**